### PR TITLE
Docker environment for run tests

### PR DESCRIPTION
**Pull request by ziollek**
I've prepared Dockerfile which enable to run tests in isolated environment with all needed tools. I hope it helps newbies like me to verify their PR.

While i was running 
```
make longtest 
```

i noticed some undeterministic behaviour (some tests ie. auth_https.tdir sometimes failed due to connection timeout to petal).
I found some problem in **util/netevent.c** which cause switching to reading phase before send http request due to incomplete ssl_handshake.

### CODE REVIEW

File: .gitignore
Comment (gthess) [inline .gitignore]: I would keep these (editor specific ignores) in your [local (global) git configuration](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreexcludesFile).
We like to populate the `.gitignore` file only with unbound related files.

File: util/iana_ports.inc
Comment (gthess) [inline util/iana_ports.inc]: I would prefer to leave this change to the maintainers.
Updating IANA ports is one of the steps leading to a release.

File: configure
Comment (gthess) [inline configure]: These changes need to go to `configure.ac` instead. `configure` is auto-generated (also) by the maintainers based on that file.

File: doc/README.tests
Comment (gthess) [inline doc/README.tests]: This would need to be updated for the new Dockerfile location.

File: util/netevent.c
Comment (gthess) [inline util/netevent.c]: With this condition now, the code that follows turns to dead code.
I would rewrite as:
```
if(want == SSL_ERROR_WANT_READ) {
			/* According to https://www.openssl.org/docs/man1.1.1/man3/SSL_do_handshake.html
			 * we should repeat handshake - for non blocking BIO
			 */
			if(c->ssl_shake_state == comm_ssl_shake_write) {
			    comm_point_listen_for_rw(c, 0, 1);
			}
			return 1;
```

The same logic, adjusted, can also be used for the the `SSL_ERROR_WANT_WRITE` case below.

File: Dockerfile
Comment (gthess) [inline Dockerfile]: I would prefer this file to be renamed to `Dockerfile.tests` and also moved under the `contrib/`directory.
Otherwise it gives the impression that it is the main Dockerfile to run unbound.

**Review by gthess — CHANGES_REQUESTED** [review]
Hi, thanks for taking the time for this!

I have some comments for changes before being able to merge.
If something is not clear let me know.

I was hunting an ssl bug in windows and came to the same conclusion for the ssl handshake. May I ask how you stumbled upon this?

File: util/netevent.c
Comment (gthess) [reply util/netevent.c]: We had a better look for this change and we solved our case (https on windows) with commit https://github.com/NLnetLabs/unbound/commit/ca67691092127aafd3c11ead622dcb659959e1bf.
Can you try that one and see if it also solves your use case?

Btw the ssl handshake is (re)tried on both read and write events until the c->ssl_shake_state reaches comm_ssl_shake_none.

File: util/iana_ports.inc
Comment (ziollek) [reply util/iana_ports.inc]: I've reverted this change

File: .gitignore
Comment (ziollek) [reply .gitignore]: I've reverted this change

**Comment by ziollek** [comment]
> Hi, thanks for taking the time for this!
> 
> I have some comments for changes before being able to merge.
> If something is not clear let me know.
> 
> I was hunting an ssl bug in windows and came to the same conclusion for the ssl handshake. May I ask how you stumbled upon this?

Thank You for review my PR. I observed problem with handshake while launching longtests in docker, i started digging and found ssl connection was waiting for response despite request wasn't send due to not finished handshake.

I merged actual master to my branch and verified that change ca67691 fixed also above problem - so i've reverted my changes in netevent.c. I've adjusted my code to other suggestions. Please let me know if i can improve something else.

Best regards.

File: Dockerfile
Comment (ziollek) [reply Dockerfile]: done

File: util/netevent.c
Comment (ziollek) [reply util/netevent.c]: Thanks! This solution resolves also my use cases.

File: doc/README.tests
Comment (ziollek) [reply doc/README.tests]: done

File: configure
Comment (ziollek) [reply configure]: done

**Comment by gthess** [comment]
Thanks, looks good to me!
Currently we are in code freeze for the upcoming release but I will merge this right after that.

Something that just came to mind, I am not sure about the portability of the `-E` option to grep. In most man pages I find online I see support for it but we will find out when testing on different platforms for the release. Worst case we convert it back to regular regular expression.

**Comment by ziollek** [comment]
Thanks for information :)

**Review by gthess — APPROVED** [review]

**Pull request merged**
